prepare_data.py

```diff
@@ -30,54 +26,76 @@ def main():
 
     Arguments:
     * --input - path to the folder with all necessary files
-    * --division - women/mixed/open/all
     * --season - current year
+    * --division - women/mixed/open/all
     * --output - path to the folder to save the output CSVs
 
-    Procedure:
-    * Read the tournament results CSVs and take only the games for the given division
-    * Read the list of EUF teams; replace aliases where applicable
-    * Read teams at tournaments list; add suffix to all teams without EUF-sanctioned roster for the given tournament
-    * Process the Game Table (check for invalid rows)
-    * Calculate basic statistics for the season (without any rankings)
-    * Save the output CSVs
-
     Outputs:
     * CSVs with Games, Tournaments, Calendar, and Summary (without any rankings)
     """
-    parser = argparse.ArgumentParser(description="EUF data preparation parser.")
+    parser = argparse.ArgumentParser(description="Parser for EUF data preparation.")
     parser.add_argument("--input", required=True, type=Path, help="Path to the folder with all necessary files")
+    parser.add_argument("--season", required=True, type=int, help="Current year (for naming purposes)")
     parser.add_argument(
         "--division", default="all", choices=["women", "mixed", "open", "all"], help="Division (women/mixed/open/all)"
     )
-    parser.add_argument("--season", required=True, type=int, help="Current year (for naming purposes)")
     parser.add_argument("--output", required=True, type=Path, help="Path to the folder to save the output CSVs")
     args = parser.parse_args()
 
+
     if args.division == "all":
         divisions = ["mixed", "open", "women"]  # Prepare data for all divisions at once
     else:
         divisions = [args.division]
 
-    setup_logger(args.output / f"prepare_data-{args.season}-{args.division}.log")
+    prepare_data(args.input, args.season, divisions, args.output)
+
+def prepare_data(input_path, season, divisions, output_path):
+    """
+    Take data from all the CSV files in the input folder and join them to create a big Game Table with the clean data;
+    export some preliminary summary statistics (no rankings are calculated here).
+
+    Prerequisites:
+    * Prepare a folder with the tournament result data - CSV files with columns Tournament, Date, Team_1, Team_2,
+      Score_1, Score_2, and Division
+    * Add to the same folder a TXT file per division specifying the teams in the EUF system; multiple aliases can be
+      defined for each team in the same row, separated with commas (filename should be teams-<division>.txt)
+    * Add to the same folder a TXT file with the pairs <team>, <tournament>; specifying that the given team has met the
+      EUF roster requirements for the particular tournament (filename should be teams_at_tournaments-<division>.txt)
+
+    Procedure:
```

Comment:
You list all the logical steps performed by the function.
This is a strong indicator, that the the whole code should be split up into exactly these steps as individual functions.
But this is probably not in scope of this PR. Maybe not even worth it.

Reply:
Generally the approach is not wrong, but I would say for ~100 lines of code it is definitely not worth it.

Reply:
this is just copy and paste to keep with the way it was done in the other files. I agree with what you're suggesting, but that's more of a dedicated change/refactoring and the purpose of this PR is to allow Ravi to enter a single command and smoothly run data preparation and ranking calculation